include/verilated.cpp

```diff
@@ -2494,9 +2494,27 @@ VerilatedContext::Serialized::Serialized() {
     m_timeprecision = picosecond;  // Initial value until overridden by _Vconfigure
 }
 
+bool VerilatedContext::assertOn() const VL_MT_SAFE {
+    const VerilatedLockGuard lock{m_mutex};
+    return m_s.m_assertOn;
+}
 void VerilatedContext::assertOn(bool flag) VL_MT_SAFE {
     const VerilatedLockGuard lock{m_mutex};
-    m_s.m_assertOn = flag;
+    // set all assert types to 'on' when true, set all types to 'off' when false
+    m_s.m_assertOn = std::numeric_limits<VerilatedAssertType_t>::max()
+                     * static_cast<VerilatedAssertType_t>(flag);
+}
+bool VerilatedContext::getAssertOn(VerilatedAssertType_t flags) const {
+    const VerilatedLockGuard lock{m_mutex};
+    return m_s.m_assertOn & flags;
+}
```

Comment:
Move the AND into where used?

Reply:
I'd replaced lock usage with atomic variable. I think it will be better to leave get,set,clear methods for assertOn to make API more consistent. Moreover, in the future more `assertcontrol` arguments will be used thus leaving separate method for each operation will make generated code more readable.